### ISSUE TITLE
Scaling issues with size parameters for image processors

### ISSUE THREAD

**Issue by milasudril** [milasudril/texpainter#5]
Currently, size parameters for image processors use a range that is dependent on the output image size. This is good from a user's perspective, but is not well suited for a hires render. For a hires render to work properly, size parameters must be proportional to the output image size. Also, if the size is used to control a filter kernel, it must be divided by the render scale factor.

- [ ] Decouple parameter values from user input. This means that more metadata is needed.

- [ ] Figure out how to handle kernel sizes. Possible solutions:
 1. Add option to scale the radius with the render size scale factor. User needs to enable this option when the generator is used as a filter mask
 2. Remove "Multiply image spectrum and grayscale image", and replace it with "Filter image spectrum". The second input will be a different type than "Grayscale image" to prevent bad scaling behavior. "Make radial gradient" will exist in two versions.

**Comment by milasudril** [comment]
Affected lines:

```
model/document.cpp:95:              auto r = 0.5 * std::exp2(brush_radius * 0.5 * std::log2(area(img)));
model/document.cpp:112:             auto r = 0.5 * std::exp2(brush_radius * 0.5 * std::log2(area(img)));
imgproc/make_radial_gradient.imgproc.md:52:     return 0.5 * std::exp2(std::lerp(-std::log2(canvas_size), 0.0, val.value()));
imgproc/checkerboard_pattern.imgproc.md:32:     return 0.5 * std::exp2(std::lerp(-std::log2(size), 0.0, val.value()));
imgproc/gingham_pattern.imgproc.md:32:  return 0.5 * std::exp2(std::lerp(-std::log2(size), 0.0, val.value()));
imgproc/grayscale_wave_distortion.imgproc.md:57:        return 0.5 * std::exp2(std::lerp(-std::log2(canvas_size), 0.0, val.value()));
```

Also for for point 2, make_topographic_map needs to be reviewed

**Comment by milasudril** [comment]
Clamp values initially, instead of adding new metadata. Clamping would probably be needed anyways to make sure that input is sane.

**Comment by milasudril** [comment]
All but make topographic map completed. Need to derive some relations here. Note. When differentiating a sharp edge, it will always be one pixel *wide*, thus the downsampled render will have a faint contour.

**Comment by milasudril** [comment]
Completed make topographic map. Note: ImageSource needs some interpolation for differentiation to work properly in hires mode.

**Comment by milasudril** [comment]
Linear interpolation in ImageSource implemented in 851ab5fbac865092bf83cade31be6cdbf84ee83c
